common-lib/src/main/scala/com/gu/mediaservice/lib/config/CommonConfig.scala

```diff
@@ -46,18 +46,32 @@ trait CommonConfig {
   val isProd: Boolean = stage == "PROD"
   val isDev: Boolean = stage == "DEV"
 
-  final val thrallKinesisStream = s"$stackName-thrall-$stage"
+  final val thrallKinesisStream = properties("thrall.kinesis.stream.name")
```

Comment:
We'd want this to be a `lazy val` rather than a `final`.

The reasoning is a bit silly... `lazy` means it's only executed when accessed. As the [test creates a ThrallConfig](https://github.com/guardian/grid/blob/master/thrall/test/lib/ElasticSearch6Test.scala#L13), `final` will mean the test is trying to find this property in the config file `/etc/gu/thrall.properties` which it won't find as the file doesn't exist and so the build fails. It's silly because the test isn't even using `ThrallConfig` for anything, its just part of the class that's being tested.

Ideally we'd restructure that part of the code, however that probably belongs in its own PR. Moving to `lazy` allows us to be... lazy.

Reply:
If we rebase off guardian/master, we can run `scripts/ci/grid.sh` to witness what CI is doing.

Reply:
Ah that makes sense. I have created a PR to merge into this branch with the fix (I've lost push access to this branch i think due to the rebase)

Reply:
I've approved and merged the fix

Reply:
We should consider adding something that allows us to approve CI to run on branches. [Akka has a good model](https://github.com/akka/akka/pull/26770#issuecomment-484316945) where one of the core team can comment on a PR to have it built and tested

Reply:
Thanks! I'm a bit confused as to what's happened but the branch seems to have moved to here now: https://github.com/guardian/grid/tree/bbc_feature/configurable-hardcoded-properties so I am assuming we should close this PR, and open a PR there?

Reply:
> We'd want this to be a `lazy val` rather than a `final`.
> 
> The reasoning is a bit silly... `lazy` means it's only executed when accessed. As the [test creates a ThrallConfig](https://github.com/guardian/grid/blob/master/thrall/test/lib/ElasticSearch6Test.scala#L13), `final` will mean the test is trying to find this property in the config file `/etc/gu/thrall.properties` which it won't find as the file doesn't exist and so the build fails. It's silly because the test isn't even using `ThrallConfig` for anything, its just part of the class that's being tested.
> 
> Ideally we'd restructure that part of the code, however that probably belongs in its own PR. Moving to `lazy` allows us to be... lazy.

https://github.com/guardian/grid/pull/2548 should help resolve this.

Will look at rebasing #2546 once that's gone in.